experiment_prototype/scan_classes/sequences.py

```diff
@@ -228,22 +228,23 @@ def make_pulse_dict(pulse_timing_info):
         # (numberofreceivesamples) using scope sync and send that to the driver to sample at
         # a specific rxrate (given by the config).
 
-        # number of samples for the first range for all slice ids
+        # number ranges to the first range for all slice ids
 
         range_as_samples = lambda x,y: int(math.ceil(x/y))
-        first_range_samples = {slice_id : range_as_samples(self.slice_dict[slice_id]['first_range'],
+        num_ranges_to_first_range = {slice_id : range_as_samples(self.slice_dict[slice_id]['first_range'],
                                                             self.slice_dict[slice_id]['range_sep'])
                                 for slice_id in self.slice_ids}
 
         # time for number of ranges given, in us, taking into account first_range and num_ranges.
-        self.ssdelay = max([(self.slice_dict[slice_id]['num_ranges'] + first_range_samples[slice_id]) *
-                            (1.0e6/self.transmit_metadata['output_rx_rate']) for slice_id in self.slice_ids])
-
+        # pulse_len is the amount of time for any range.
+        self.ssdelay = max([(self.slice_dict[slice_id]['num_ranges'] + num_ranges_to_first_range[slice_id]) *
+                            self.slice_dict[slice_id]['pulse_len'] for slice_id in self.slice_ids])
```

Comment:
Does pulse_len need to be converted to seconds here?

Reply:
so finally checked this - this value is in us (it's also documented as being in us) and pulse_len is also in us so I think it's correct. It used to be 1.0e6/output_rx_rate which would've also given us